src/merkle/store/mod.rs

```diff
@@ -280,6 +277,33 @@ impl<T: KvMap<RpoDigest, StoreNode>> MerkleStore<T> {
         })
     }
 
+    /// Extracts a [MerkleDelta] object by traversing the two trees specified by their roots and
+    /// depth and comparing the leaves.
+    pub fn merkle_tree_delta(
+        &self,
+        tree_root_1: RpoDigest,
+        tree_root_2: RpoDigest,
+        depth: u8,
+    ) -> Result<MerkleDelta, MerkleError> {
```

Comment:
I'm not 100% sure about this method. It feels very specialized for one specific use case and it probably won't work for things like TSMT. So, we'd have to add another method for TSMT (and potentially other use cases) and it feels like we are polluting the Merkle store interface somewhat.

I'm wondering if a more generic approach would work better here. For example, we could have a method which would iterate over non-empty leaves. It could look something like this:
```Rust
pub fn non_empty_leaves(&self, root: RpoDigest, max_depth: u8) -> LeafIterator
```
This iterator could yield `(NodeIndex, RpoDigest)` values, skipping all empty leaves.

Then, users of `MerkleStore` would be able to iterate over leaves of two trees and build deltas for them as needed.